### ISSUE TITLE
App doesn't create articles after extraction error

### ISSUE THREAD

**Issue by frankjdelgado** [frankjdelgado/wiki-history-extractor-api#33]
TL;DR: Extract the article first and then the revisions. `get_article()` must be inside `get_one()` just in case. Optimize code on `get_article()` or `get_one()` if necessary

Current workflow:

1.  User requests extractions of an article

2. Apps delegate to celery worker

3. Worker extracts first chunk of revisions

4. Worker queries for the article page, save article to DB

If an error occurs during step 3, step 4 doesn't happens and the database will, sometimes, have revisions without its corresponding article document

Desired workflow after step 2:

3. Check if article exists in DB
  3.1 If not, extract article, save to DB and continue
  3.2 If exists, update last extraction date

4. Extract revisions
